### ISSUE TITLE
Is it possible to grant permission for Termux to use loop devices?

### ISSUE THREAD

**Issue by sdrausty** [termux/termux-app#375]
Is it possible to grant permission for Termux to use loop devices like we do with the storage **slider** in Settings > Apps > Termux? Or can this feature be added without one? Can this feature be added at all?

![screenshot_20170801-135615](https://user-images.githubusercontent.com/27742457/28839336-4eeabe50-76c1-11e7-8e8a-8d2dc61cb094.png)

```
$ /system/bin/make_ext4fs -l 10M file
Creating filesystem with parameters:
    Size: 10485760
    Block size: 4096
    Blocks per group: 32768
    Inodes per group: 640
    Inode size: 256
    Journal blocks: 1024
    Label:
    Blocks: 2560
    Block groups: 1
    Reserved block group size: 7
Created filesystem with 11/640 inodes and 1078/2560 blocks
$ /system/bin/toolbox mount -o loop -t ext4 file mountDir/
open loop device failed: Permission denied
$ /system/bin/toybox mount -o loop -t ext4 file mountDir
mount: bad /etc/fstab: No such file or directory
$ /system/bin/mount -o loop -t ext4 file mountDir/
open loop device failed: Permission denied
$ /sys/fs/selinux/class/filesystem/perms/mount -o loop -t ext4 file mountDir/
bash: /sys/fs/selinux/class/filesystem/perms/mount: Permission denied
```

In trying to create more native space on device for Termux, loop device access is a stumbling point! "Does anyone want to have more native Android space on device? #1176" https://github.com/termux/termux-packages/issues/1176

https://android.stackexchange.com/questions/180537/what-is-the-source-code-for-the-intent-or-permission-for-an-android-app-to-use-l

^_^My fist post about this issue issue is long ago deleted from stackexchange. The post above has been put on hold @present.^_^

Will the following links help in resolving this issue about intents and android permissions for grant_loop_device or similar?

If Releases · meefik/linuxdeploy · GitHub **can mount a filesystem from a file, Termux can too** https://github.com/meefik/linuxdeploy

Old links that may still be rellavent:

http://loopback.io/doc/en/lb3/Android-SDK.html

 https://www.google.com/search?q=intent+permission+android+loopback

**What should Termux be able to do?

Mount a filesystem from a file using something similar to what Linux Deploy uses for native Android filespace aquasition.**

**Comment by sdrausty** [comment]
Feature Request: `mount loopback device`

**Comment by sdrausty** [comment]
__Evidently, we can launch activities without root by passing the --user 0 flag to am start:__ from https://github.com/termux/termux-app/issues/206

**Comment by neoslaughter** [comment]
Creating a loop device needs root access as far as I know.
